CoolCurrency/Repository/DatabaseRepository.swift

```diff
@@ -49,6 +50,50 @@ class DatabaseRepository: DatabaseRepositable {
         })
     }
     
+    func updateUsersScoreboard(SignedInUser userNumber: Int, name userName: String, finalScore userFinalScore: String, totalScore userTotalScore: String, completion: @escaping DatabaseResponse) {
+        let userObject: [String: Any] = [
+            "Name": userName as NSObject,
+            "FinalScore": userFinalScore,
+            "TotalScore": userTotalScore
+        ]
+        
+        self.databaseReference.child("Scoreboard/UserNumber\(userNumber)").setValue(userObject) { (error: Error?, _: DatabaseReference) in
+            if let error = error {
+                completion(.failure(error))
+            } else {
+                completion(.success(true))
+            }
+        }
+    }
+    
+    func retrieveUserScoreboards(completion: @escaping(Result<[LeadershipBoardDataModel], Error>) -> Void) {
+        var leaderBoardsList: [LeadershipBoardDataModel] = []
+        let completionLeaderBoardItem = DispatchWorkItem { completion(.success(leaderBoardsList)) }
+        databaseReference.child("Scoreboard").observeSingleEvent(of: .value, with: { (snapshot) in
+            let numberOfUsers = snapshot.childrenCount
+            for userNumber in 0..<numberOfUsers {
+                self.dispatchGroup.enter()
+                self.databaseReference.child("Scoreboard/UserNumber\(userNumber)").observeSingleEvent(of: .value) { (snapshot) in
+                    let value = snapshot.value as? NSDictionary
+                    let username = value?["Name"] as? String ?? ""
+                    let correctAnswers = value?["FinalScore"] as? String ?? ""
+                    let totalScores = value?["TotalScore"] as? String ?? ""
+                    let tempLeadershipBoard = LeadershipBoardDataModel(userNumber: Int(userNumber),
+                                                                       name: username,
+                                                                       correctAnswers: Int(correctAnswers) ?? 0,
+                                                                       totalScore: Int(totalScores) ?? 0)
+                    leaderBoardsList.append(tempLeadershipBoard)
+                    self.dispatchGroup.leave()
+                }
+            }
+            self.dispatchGroup.notify(queue: DispatchQueue.main, work: completionLeaderBoardItem)
+            // swiftlint:disable multiple_closures_with_trailing_closure
```

Comment:
So just something to think about, but you can update the swiftLint yaml and customise the rules, then you won't need to silence the warnings 😄 consider that with your next PR then you can go through and clean these comments up.